File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizBiometricSheet.swift
import SwiftUI

public struct BizBiometricSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    var type: String // "both", "face", "fingerprint"
    var onSetup: ((String) -> Void)?
    var onLater: (() -> Void)?
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            content: {
                VStack(spacing: 0) {
                    // Shield Icon
                    ZStack {
                        Circle()
                            .fill(Color(hex: "#E6F0FF"))
                            .frame(width: 72, height: 72)
                        
                        Circle()
                            .fill(Color(hex: "#003399"))
                            .frame(width: 48, height: 48)
                        
                        Image(systemName: "checkmark.shield.fill")
                            .font(.system(size: 24))
                            .foregroundColor(.white)
                    }
                    .padding(.top, 8)
                    .padding(.bottom, 24)
                    
                    // Text
                    Text("Unlock faster with \(titleName)")
                        .font(.system(size: 20, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .multilineTextAlignment(.center)
                        .padding(.bottom, 12)
                    
                    Text("Skip PIN entry and securely access your account using \(subtitleName).")
                        .font(.system(size: 13))
                        .foregroundColor(Color(hex: "#555555"))
                        .multilineTextAlignment(.center)
                        .lineSpacing(4)
                        .padding(.horizontal, 16)
                        .padding(.bottom, 32)
                    
                    // Cards
                    ZStack {
                        HStack(spacing: 16) {
                            if type == "both" || type == "face" {
                                BiometricCard(
                                    icon: "faceid",
                                    title: "Face ID",
                                    desc: "Use facial recognition"
                                )
                            }
                            
                            if type == "both" || type == "fingerprint" {
                                BiometricCard(
                                    icon: "touchid",
                                    title: "Fingerprint ID",
                                    desc: "Use your fingerprint"
                                )
                            }
                        }
                        
                        if type == "both" {
                            Circle()
                                .fill(Color(hex: "#F8F9FA"))
                                .frame(width: 32, height: 32)
                                .shadow(color: Color.black.opacity(0.06), radius: 4, x: 0, y: 2)
                                .overlay(
                                    Text("Or")
                                        .font(.system(size: 12, weight: .bold))
                                        .foregroundColor(Color(hex: "#1A2A5E"))
                                )
                        }
                    }
                    .padding(.bottom, 32)
                    
                    // Info Banner
                    HStack(alignment: .top, spacing: 12) {
                        Image(systemName: "info.circle.fill")
                            .foregroundColor(Color(hex: "#0066CC"))
                            .font(.system(size: 20))
                            .padding(.top, 2)
                        
                        VStack(alignment: .leading, spacing: 4) {
                            Text("Face / Fingerprint ID")
                                .font(.system(size: 13, weight: .bold))
                                .foregroundColor(Color(hex: "#1A2A5E"))
                            
                            Text("Link your Face or Fingerprint ID for quick and easy access to your account. Enjoy seamless logins and hassle-free payments by setting it up now.")
                                .font(.system(size: 11))
                                .foregroundColor(Color(hex: "#666666"))
                                .lineSpacing(2)
                        }
                    }
                    .padding(.bottom, 32)
                    
                    // Actions
                    VStack(spacing: 12) {
                        BizButton("Set \(buttonName) ID", variant: .primary, action: {
                            onSetup?(type)
                            isPresented = false
                        })
                        
                        BizButton("Maybe Later", variant: .outline, action: {
                            onLater?()
                            isPresented = false
                        })
                    }
                    .padding(.bottom, 32)
                }
                .padding(.horizontal, 24)
            }
        )
    }
    
    private var titleName: String {
        if type == "both" { return "Biometrics" }
        if type == "face" { return "Face ID" }
        return "Fingerprint ID"
    }
    
    private var subtitleName: String {
        if type == "both" { return "Face ID or Fingerprint" }
        if type == "face" { return "Face ID" }
        return "Fingerprint"
    }
    
    private var buttonName: String {
        if type == "both" { return "Biometrics" }
        if type == "face" { return "Face" }
        return "Fingerprint"
    }
}

private struct BiometricCard: View {
    let icon: String
    let title: String
    let desc: String
    
    var body: some View {
        VStack(spacing: 0) {
            Image(systemName: icon)
                .font(.system(size: 40))
                .foregroundColor(Color(hex: "#0066CC"))
                .padding(.bottom, 16)
            
            Text(title)
                .font(.system(size: 14, weight: .bold))
                .foregroundColor(Color(hex: "#1A2A5E"))
                .padding(.bottom, 6)
            
            Text(desc)
                .font(.system(size: 11))
                .foregroundColor(Color(hex: "#777777"))
                .multilineTextAlignment(.center)
                .lineSpacing(2)
        }
        .padding(.vertical, 24)
        .padding(.horizontal, 16)
        .frame(maxWidth: .infinity)
        .background(Color.white)
        .cornerRadius(16)
        .overlay(
            RoundedRectangle(cornerRadius: 16)
                .stroke(Color(hex: "#EAEAEA"), lineWidth: 1)
        )
        .shadow(color: Color.black.opacity(0.03), radius: 6, x: 0, y: 4)
    }
}

public extension View {
    func bizBiometricSheet(
        isPresented: Binding<Bool>,
        type: String = "both",
        onSetup: ((String) -> Void)? = nil,
        onLater: (() -> Void)? = nil
    ) -> some View {
        self.modifier(BizBiometricSheetModifier(
            isPresented: isPresented,
            type: type,
            onSetup: onSetup,
            onLater: onLater
        ))
    }
}

File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizBiometricSheet.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Info
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material3.Icon
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@Composable
fun BizBiometricSheet(
    isVisible: Boolean,
    type: String = "both", // "both", "face", "fingerprint"
    onDismissRequest: () -> Unit,
    onSetup: (String) -> Unit = {},
    onLater: () -> Unit = {}
) {
    val titleName = when (type) {
        "face" -> "Face ID"
        "fingerprint" -> "Fingerprint ID"
        else -> "Biometrics"
    }

    val subtitleName = when (type) {
        "face" -> "Face ID"
        "fingerprint" -> "Fingerprint"
        else -> "Face ID or Fingerprint"
    }

    val buttonName = when (type) {
        "face" -> "Face"
        "fingerprint" -> "Fingerprint"
        else -> "Biometrics"
    }

    BizBottomSheet(
        isVisible = isVisible,
        onDismissRequest = onDismissRequest
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(bottom = 32.dp),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            // Shield Icon
            Box(
                modifier = Modifier.padding(top = 8.dp, bottom = 24.dp)
            ) {
                Box(
                    modifier = Modifier
                        .size(72.dp)
                        .background(Color(0xFFE6F0FF), CircleShape),
                    contentAlignment = Alignment.Center
                ) {
                    Box(
                        modifier = Modifier
                            .size(48.dp)
                            .background(Color(0xFF003399), CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Lock,
                            contentDescription = "Shield",
                            tint = Color.White,
                            modifier = Modifier.size(24.dp)
                        )
                    }
                }
            }

            Text(
                text = "Unlock faster with $titleName",
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                textAlign = TextAlign.Center,
                modifier = Modifier.padding(bottom = 12.dp)
            )

            Text(
                text = "Skip PIN entry and securely access your account using $subtitleName.",
                fontSize = 13.sp,
                color = Color(0xFF555555),
                textAlign = TextAlign.Center,
                lineHeight = 18.sp,
                modifier = Modifier.padding(horizontal = 16.dp).padding(bottom = 32.dp)
            )

            // Selection Cards
            Box(
                modifier = Modifier.fillMaxWidth().padding(bottom = 32.dp)
            ) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = if (type == "both") Arrangement.SpaceBetween else Arrangement.Center
                ) {
                    if (type == "both" || type == "face") {
                        Surface(
                            modifier = Modifier
                                .weight(if (type == "both") 1f else 0f)
                                .width(if (type != "both") 160.dp else 0.dp),
                            shape = RoundedCornerShape(16.dp),
                            color = Color.White,
                            border = BorderStroke(1.dp, Color(0xFFEAEAEA)),
                            shadowElevation = 2.dp
                        ) {
                            Column(
                                modifier = Modifier.padding(vertical = 24.dp, horizontal = 16.dp),
                                horizontalAlignment = Alignment.CenterHorizontally
                            ) {
                                // Simplified icon for Face ID
                                Icon(
                                    imageVector = Icons.Default.Info, // Use appropriate icon resource
                                    contentDescription = "Face ID",
                                    tint = Color(0xFF0066CC),
                                    modifier = Modifier.size(40.dp).padding(bottom = 16.dp)
                                )
                                Text("Face ID", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1A2A5E))
                                Text("Use facial recognition", fontSize = 11.sp, color = Color(0xFF777777), textAlign = TextAlign.Center)
                            }
                        }
                    }

                    if (type == "both") {
                        Spacer(modifier = Modifier.width(16.dp))
                    }

                    if (type == "both" || type == "fingerprint") {
                        Surface(
                            modifier = Modifier
                                .weight(if (type == "both") 1f else 0f)
                                .width(if (type != "both") 160.dp else 0.dp),
                            shape = RoundedCornerShape(16.dp),
                            color = Color.White,
                            border = BorderStroke(1.dp, Color(0xFFEAEAEA)),
                            shadowElevation = 2.dp
                        ) {
                            Column(
                                modifier = Modifier.padding(vertical = 24.dp, horizontal = 16.dp),
                                horizontalAlignment = Alignment.CenterHorizontally
                            ) {
                                // Simplified icon for Fingerprint
                                Icon(
                                    imageVector = Icons.Default.Info, // Use appropriate icon resource
                                    contentDescription = "Fingerprint",
                                    tint = Color(0xFF0066CC),
                                    modifier = Modifier.size(40.dp).padding(bottom = 16.dp)
                                )
                                Text("Fingerprint ID", fontSize = 14.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1A2A5E))
                                Text("Use your fingerprint", fontSize = 11.sp, color = Color(0xFF777777), textAlign = TextAlign.Center)
                            }
                        }
                    }
                }

                if (type == "both") {
                    Box(
                        modifier = Modifier
                            .align(Alignment.Center)
                            .size(32.dp)
                            .background(Color(0xFFF8F9FA), CircleShape),
                        contentAlignment = Alignment.Center
                    ) {
                        Text("Or", fontSize = 12.sp, fontWeight = FontWeight.SemiBold, color = Color(0xFF1A2A5E))
                    }
                }
            }

            // Info Banner
            Row(
                modifier = Modifier.fillMaxWidth().padding(bottom = 32.dp),
                verticalAlignment = Alignment.Top,
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.Info,
                    contentDescription = "Info",
                    tint = Color(0xFF0066CC),
                    modifier = Modifier.size(20.dp).padding(top = 2.dp)
                )
                Column {
                    Text("Face / Fingerprint ID", fontSize = 13.sp, fontWeight = FontWeight.Bold, color = Color(0xFF1A2A5E), modifier = Modifier.padding(bottom = 4.dp))
                    Text(
                        "Link your Face or Fingerprint ID for quick and easy access to your account. Enjoy seamless logins and hassle-free payments by setting it up now.",
                        fontSize = 11.sp,
                        color = Color(0xFF666666),
                        lineHeight = 16.sp
                    )
                }
            }

            BizButton(
                text = "Set $buttonName ID",
                variant = BizButtonVariant.Primary,
                block = true,
                onClick = {
                    onSetup(type)
                    onDismissRequest()
                }
            )
            Spacer(modifier = Modifier.height(12.dp))
            BizButton(
                text = "Maybe Later",
                variant = BizButtonVariant.Outline,
                block = true,
                onClick = {
                    onLater()
                    onDismissRequest()
                }
            )
        }
    }
}
